index.js

```diff
@@ -233,16 +234,47 @@ function reflowText (text, width, gfm) {
   return reflowed.join('\n');
 }
 
-function indentLines (text, tab) {
-  return text.replace(/\n/g, '\n' + tab) + '\n\n';
+function indentLines (indent, text) {
+  return text.replace(/(^|\n)(.+)/g, '$1' + indent + '$2');
 }
 
-function changeToOrdered(text) {
-  var i = 1;
-  return text.split('\n').reduce(function (acc, line) {
-    if (!line) return '\n' + acc;
-    return acc + line.replace(/(\s*)\*/, '$1' + (i++) + '.') + '\n';
-  });
+function indentify(indent, text) {
+  if (!text) return text;
+  return indent + text.split('\n').join('\n' + indent);
+}
+
+var listItemPoint = /^\s*(\*|\d+\.)/;
+
+function bulletPointLine (line) {
+  return line.match(listItemPoint) ? line : '* ' + line;
+}
+
+function bulletPointLines (lines) {
+  return lines.split('\n')
+    .filter(identity)
+    .map(bulletPointLine)
+    .join('\n');
+}
+
+function numberedLine (line, num) {
+  return line.match(listItemPoint) ? line : (num+1) + '. ' + line;
+}
+
+function numberedLines (lines) {
+  return lines.split('\n')
+    .filter(identity)
+    .map(numberedLine)
+    .join('\n');
+}
+
+function list(body, ordered) {
+  body = body.trim();
+  body = ordered ? numberedLines(body) : bulletPointLines(body);
+  return body;
+}
+
+function section (text) {
+  return '\n' + text + '\n';
```

Comment:
If you change this to `return text + '\n\n';` you will see that everything works, the newline at the start of output is gone, but nested lists's first item will be merged into the last item of it's parent list.